src/lib/components/fileUploader/fileUploader.js

```diff
@@ -0,0 +1,234 @@
+import Uppy from "@uppy/core";
+import AwsS3 from "@uppy/aws-s3";
+import Webcam from "@uppy/webcam";
+import { DashboardModal, Dashboard } from "@uppy/react";
+import { checkValidations } from "../../utils/validation";
+import {
+  componentPropTypes,
+  defaultComponentPropTypes,
+} from "../../utils/componentPropTypes";
+
+import styles from "./fileUploader.module.scss";
+
+class FileUploader extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalOpen: false,
+      validationErrors: "",
+    };
+    this.uppy = Uppy({
+      restrictions: {
+        maxNumberOfFiles: props.maxNumberOfFiles,
+        allowMultipleUploads: props.allowMultipleUploads,
+        autoProceed: props.autoProceed,
+        maxFileSize: props.maxFileSize || undefined,
+        minNumberOfFiles: props.minNumberOfFiles || undefined,
+        allowedFileTypes: props.allowedFileTypes || undefined,
+      },
+      onBeforeUpload: files => {
+        return Object.keys(files).reduce((accum, key) => {
+          if (!files[key].meta.isUploaded) {
+            accum[key] = files[key];
+          }
+          return accum;
+        }, {});
+      },
+    });
+    this.uppy.use(Webcam);
+    this.uppy.use(AwsS3, {
+      companionUrl: props.companionUrl,
+      // serversHeader I couldn't get them into the server
+      serverHeaders: {
+        folderName: props.folderInS3Name,
+      },
+    });
+    this.uppy.on("upload-success", (file, data) =>
+      this.handleUploadFiles(file, data)
+    );
+    this.uppy.on("cancel-all", () => this.handleUploadFiles());
+    this.uppy.on("file-added", file => {
+      if (file.meta.isUploaded) {
+        this.uppy.setFileState(file.id, {
+          progress: { uploadComplete: true, uploadStarted: false },
+        });
+      }
+    });
+
+    const filesUploadedArray = this.getFilesUploadedAsArray(this.props);
+    if (filesUploadedArray.length) {
+      filesUploadedArray.forEach(file => {
+        this.uppy.addFile({
+          name: file,
+          type: "",
+          data: {},
+          meta: {
+            isUploaded: true,
+          },
+          source: "Local", // optional, determines the source of the file, for example, Instagram
+          isRemote: true, // optional, set to true if actual file is not in the browser, but on
+          // some remote server, for example, when using companion in combination with Instagram
+        });
+      });
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { validations, filesUploaded, hasErrorAfterSubmit } = this.props;
+    if (hasErrorAfterSubmit !== prevProps.hasErrorAfterSubmit) {
+      const validationErrors = checkValidations(validations, filesUploaded);
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ validationErrors });
+    }
+  }
+
+  getFilesUploadedAsArray = props => {
+    if (props.filesUploaded && props.filesUploaded.length) {
+      return props.filesUploaded.split(", ");
```

Comment:
Is the space after the `,` intentional? `split(", ");` I think it is but just wanted to check 😃

Reply:
yes, it is...haha 😄